src/fuzzy.c

```diff
@@ -933,32 +961,36 @@ fuzzymatches_to_strmatches(
     static int
 has_match(char_u *needle, char_u *haystack)
 {
-    while (*needle != NUL)
+    if (!needle || !haystack || !*needle)
+	return !*needle;
```

Comment:
that doesn't look correct. If needle is null, this crashes. Also if *needle is zero, this returns true, so a match?